### PR TITLE
Update concept paths

### PR DESCRIPTION
**Pull request by forus**
This PR is based on [the single transaction PR](https://github.com/thehyve/transmart-core/pull/149).

### CODE REVIEW

**Comment by jochemb** [comment]
I welcome this addition!

File: transmart-copy/src/main/groovy/org/transmartproject/copy/table/Concepts.groovy [hunk 1]
```diff
@@ -93,9 +97,24 @@ class Concepts {
...
+                                oldToNewConceptPath[conceptCodeToConceptPath[conceptCode]] = conceptPath
+                                conceptCodeToConceptPath[conceptCode] = conceptPath
+                                updatedCount++
+                            } else {
+                                log.error "Error: trying to load concept with code ${conceptCode} and path ${conceptPath},"
+                                log.error "but concept with that code already exists with path ${conceptCodeToConceptPath[conceptCode]}."
+                                throw new IllegalStateException("Cannot load concept with code ${conceptCode}. Other concept already exists with that code.")
+                            }
                         }
                         existingCount++
```
Comment (gijskant) [inline transmart-copy/src/main/groovy/org/transmartproject/copy/table/Concepts.groovy]: add `else`?

File: transmart-copy/src/main/groovy/org/transmartproject/copy/table/TreeNodes.groovy [hunk 2]
```diff
@@ -152,4 +153,32 @@ class TreeNodes {
         }
     }
 
+    private updateTreeNodePaths(Map<String, String> oldToNewConceptPath) {
```
Comment (gijskant) [inline transmart-copy/src/main/groovy/org/transmartproject/copy/table/TreeNodes.groovy]: Aren't we now again hardcoding conventions about concept path and tree path to be the same?

File: transmart-copy/src/test/groovy/org/transmartproject/copy/CopySpec.groovy [hunk 3]
```diff
@@ -74,52 +74,38 @@ class CopySpec extends Specification {
...
-        copy.deleteStudy(TEST_STUDY)
+        copy.deleteStudyById(TEST_STUDY)
 
-        studyIds = readFieldsFromDb(copy.database, Studies.study_table, 'study_id')
-        Map afterCounts = count(copy.database, studySpecificTables)
+        Map afterCounts = count(studySpecificTables)
         Map expectedCounts = fileRowCounts
                 .collectEntries { Table table, Number rows -> [table, beforeCounts[table] - rows] }
 
         then: 'Expect the study to be loaded'
```
Comment (gijskant) [inline transmart-copy/src/test/groovy/org/transmartproject/copy/CopySpec.groovy]: not to be loaded

File: transmart-copy/src/test/groovy/org/transmartproject/copy/CopySpec.groovy [hunk 4]
```diff
@@ -162,11 +135,107 @@ class CopySpec extends Specification {
...
+        ensureTestStudyLoaded()
+        Options options = new Options()
+        options.addOption(new Option('d', 'delete', true, ''))
+
+        when: 'error during load happens'
+        CommandLine cli1 = new DefaultParser()
+                .parse(options, ['--delete', TEST_STUDY] as String[])
+        Copy.runCopy(cli1, DATABASE_CREDENTIALS)
+        then: 'study has been deleted'
+        !getTestStudyDbIdentifier()
```
Comment (gijskant) [inline transmart-copy/src/test/groovy/org/transmartproject/copy/CopySpec.groovy]: The type of `getTestStudyDbIdentifier` is `Number`. If the study has id `0`, this fails?

**Pull request merged**